### ISSUE TITLE
NVDA stops responding to key commands sometimes in Firefox

### ISSUE THREAD

**Issue by XLTechie** [nvaccess/nvda#11546]
### Steps to reproduce:

This happens from time to time.  It has happened to me in every version of NVDA and Firefox since 2018 or earlier, and people have reported it on the user list happening outside of Firefox.
However, I have no idea how to make it happen.

### Actual behavior:

Abruptly, usually starting in browse mode, or when it should have entered browse mode, NVDA will stop acting on its own gestures.  Even NVDA+N has no effect.

In the case represented by my attached log, after a while of that, it suddenly acquires the characteristic of including the Windows key with every other key (not just NVDA gestures).  This only happened after I pressed Windows+R on purpose.  After that, Windows persists.
I've never noticed that quirk before, but in this case I only noticed it by looking at the log. Surprisingly, holding down Windows didn't really interfere with most of what I tried to type.

### Expected behavior:

The keyboard keeps working normally; NVDA continues acting on gestures; the Windows key doesn't get virtually held down.

#### NVDA installed/portable/running from source:

Installed.

#### NVDA version:

alpha-20809,f4cb0847

#### Windows version:

10 Version 2004 (OS Build 19041.450)

#### Name and version of other software in use when reproducing the issue:

Firefox 80.0 (64 bit), but this has happened with versions at least as far back as the 60s.

#### Does the issue still occur after restarting your computer?

Restarting NVDA fixes the problem, so it follows that restarting Windows would also.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

Every version since 2018.X at least, maybe further back.

#### If addons are disabled, is your problem still occuring?

Hard to say.

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?

No, although I had just run it before this run of NVDA, and then did a system restart.
Restarting NVDA solved the problem, so by then it was pointless to try.
I didn't think to use the NVDA systray icon to run the COM Reg Fix while the problem was happening.

### Log:

In [this log](https://github.com/nvaccess/nvda/files/5166800/nvda.stops.acting.on.key.events.txt), I have some notes about what was going on at various times, and when I have removed sections.  The more important areas are:

```
-- Starting in Firefox --
```
And:
```
-- Windows gets stuck. --
```

**Comment by Qchristensen** [comment]
As you noted in the log, at one point, eery keystroke includes the Windows key.  I have had keyboards in the past where they seemed to get a modifier key stuck down (shift, alt or control, etc) - in that case, what I did was very technical, but basically I just kept pressing the offending key quickly  and repeatedly until say a dozen times in a couple of seconds and it usually sorted itself out.  If you try that with the Windows key when this happens does it make a difference?

Otherwise, a log with add-ons disabled would really help confirm this is NOT caused by an add-on.

**Comment by ABuffEr** [comment]
> basically I just kept pressing the offending key quickly and repeatedly until say a dozen times in a couple of seconds and it usually sorted itself out.

In my experience, pressing the two ctrl, or the two shift, etc, at same time can help too.

**Comment by XLTechie** [comment]
This was never predictable, and has become much less frequent in recent versions. The COM Reg Tool never helped with this.

That said, given all the changes in Firefox recently, I'm willing to close this until I or someone else can verify that it still happens.
